src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java

```diff
@@ -9,54 +9,124 @@
 import java.time.Duration;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * The SchemaErrors class manages schema error caching for different connections.
+ * It uses Caffeine cache to store and retrieve schema errors based on connection and schema IDs.
+ *
+ * <p>This class is application-scoped and observes connection lifecycle events to manage the cache.
+ * It provides methods to read, write, and clear schema errors for specific connections.
+ *
+ * <p>Configuration properties:
+ * <ul>
+ *   <li>`ide-sidecar.schema-fetch-error-ttl`: The time-to-live duration for schema fetch errors in the cache.</li>
+ * </ul>
+ */
 
 @ApplicationScoped
 public class SchemaErrors {
 
+  /**
+   * Represents a unique connection identifier.
+   */
   public record ConnectionId(String id) {
 
   }
 
+  /**
+   * Represents a unique schema identifier within a cluster.
+   */
   public record SchemaId(String clusterId, int schemaId) {
 
   }
 
+  /**
+   * Represents an error message associated with a schema.
+   */
   public record Error(String message) {
 
   }
 
-  private static final Duration SCHEMA_FETCH_ERROR_TTL = Duration.ofSeconds(30);
+  public static final Duration schemaFetchErrorTtl = Duration.ofSeconds(
+      ConfigProvider
+          .getConfig()
+          .getValue(
+              "ide-sidecar.schema-fetch-error-ttl",
+              Long.class));
 
-  public static final Map<ConnectionId, Cache<SchemaId, SchemaErrors.Error>> cacheOfCaches = new ConcurrentHashMap<>();
+  public static final Map<ConnectionId, Cache<SchemaId, Error>> cacheOfCaches = new ConcurrentHashMap<>();
 
-  public static Cache<SchemaId, Error> getSubCache(ConnectionId key) {
+  /**
+   * Retrieves or creates a sub-cache for a specific connection.
+   *
+   * @param key The connection identifier.
+   * @return The sub-cache for the connection.
+   */
+  public Cache<SchemaId, Error> getSubCache(ConnectionId key) {
```

Comment:
Why is this method public? IIUC, it's not used outside of this class, so we can do this:
```suggestion
  private Cache<SchemaId, Error> getSubCache(ConnectionId key) {
```
If it were used in testing, we could make it _package protected_.